scripts/leapfrog/generate_resources.py

```diff
@@ -0,0 +1,322 @@
+#! /usr/bin/env python
+from argparse import ArgumentParser
+from ipaddress import IPv4Network
+from itertools import chain, izip, repeat, islice
+from json import dumps
+import os
+from time import sleep
+
+from keystoneauth1.identity import v3
+from keystoneauth1 import session as session_
+from cinderclient import client as cinder_client
+from glanceclient import client as glance_client
+from neutronclient.v2_0 import client as neutron_client
+from novaclient import client as nova_client
+
+CLIENT_VERSION = "2"
+
+
+def get_conf():
+    parser = ArgumentParser()
+    parser.add_argument("--servers", type=int, default=1)
+    parser.add_argument("--volumes", type=int, default=0)
+    parser.add_argument("--networks", type=int, default=1)
+    parser.add_argument("--infinite-quotas", action="store_true")
+    args = parser.parse_args()
+    if args.servers > args.networks:
+        raise ValueError(
+            "Each server must have at least one network and so the number of "
+            "networks must be greater than or equal to the number of servers."
+        )
+    if args.servers < 1:
+        raise ValueError(
+            "The minimum number of servers that can be created is 1."
+        )
+
+    conf = {
+        "auth": {},
+        "infinite_quotas": args.infinite_quotas,
+        "resources": {},
+    }
+    conf["auth"]["endpoint_type"] = os.environ["OS_ENDPOINT_TYPE"]
+    conf["auth"]["username"] = os.environ["OS_USERNAME"]
+    conf["auth"]["password"] = os.environ["OS_PASSWORD"]
+    conf["auth"]["project_name"] = os.environ["OS_PROJECT_NAME"]
+    conf["auth"]["auth_url"] = os.environ["OS_AUTH_URL"]
+    conf["auth"]["user_domain_id"] = os.environ["OS_USER_DOMAIN_NAME"]
```

Comment:
These settings are placed in the openrc file, I didn't see a comment there qualifying their use. What would you hope a comment would say?